packages/drivers/routerlicious-driver/src/documentService.ts

```diff
@@ -277,9 +282,20 @@ export class DocumentService
 				summarizeProtocolTree: shouldSummarizeProtocolTree,
 			};
 
+			connection.once("disconnect", (reason: IAnyDriverError) => {
+				if (reason.errorType === RouterliciousErrorTypes.authorizationError) {
+					this.refreshTokenOnConnect = true;
```

Comment:
I think we rely here on a fact that this event handler instance will be called first, before DM's event handler has a chance to process same event and call back into connectToDeltaStream(). Does EventEmitter provide a guarantee in which order it invokes handlers? If not, we might be for a surprise here.

Reply:
That said, how much does it matter? I.e. I'd assume if token expired, it will be just a bit slower to reconnect - we will try with old token, should fail and should refresh it. As long as that sequence works correctly, we should be fine.

Reply:
Yes, the listeners are triggered in order they are registered. So, it is fine.
However, I also thought about the second point, that it will naturally work correctly beside another fail before succeeding. So, I end up, adding that prevention but can remove it too, since it will work correctly.